### PR TITLE
[Product sharing] Include product name and description for better AI generation

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Part of: #9867

## Description

[During code review](https://github.com/woocommerce/woocommerce-ios/pull/9908#pullrequestreview-1471378699), it has been identified that the prompt is not valid/matching for some products. She also noticed that some generated messages have the URL included.

This PR attempts to improve the message generation by,
- Adding the product name and description to the prompt
- Specifying not to include the URL
- Specifying not to use quotes and instead provide the raw string 

## Testing instructions

- Log in to a public WPCom site.
- Select a published product and tap Share.
- On the AI view, select Write it for me. 
- After generating a text, you can tap Regenerate to generate different text.
- Validate that the generated text doesn't have the product URL and isn't using quotation marks.

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/524475/51022eed-693c-4054-931b-502a828706ad

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by itsmeichigo — COMMENTED** [review]
It seems to me that the prompt not to include quotes doesn't work all the time. I got 2 messages with quotes within 5 times testing the same product:

 

Should we remove that prompt and manually trim the quotation marks from the results?

**Comment by selanthiraiyan** [comment]
Thanks for the review and testing @itsmeichigo 

> Should we remove that prompt and manually trim the quotation marks from the results?

I agree. 
- I will work on adding this trimming code to [this method](https://github.com/woocommerce/woocommerce-ios/pull/9921/files#diff-2e6147f83e5c53fe36dba042353e12fabda27770f2f8e360b006eccdef95e77fR560). 
- Do you think we should keep the command `- Do not quote the message and give me the raw string.` in the prompt, or should we remove it?

**Comment by itsmeichigo** [comment]
> Do you think we should keep the command - Do not quote the message and give me the raw string. in the prompt, or should we remove it?

Since this doesn't work, I would remove it. If we could keep the prompt concise, we may have better results, just in case some requirements are ignored in favor of others.

**Comment by selanthiraiyan** [comment]
Good point about keeping the prompt concise. I have pushed the necessary updates. The PR is ready for another look. 🙏

File: Yosemite/Yosemite/Stores/ProductStore.swift [hunk 1]
```diff
@@ -539,18 +539,26 @@ private extension ProductStore {
...
             "- The length should be up to 3 sentences",
             "- Try to make shorter sentences, using less difficult words to improve readability",
-            "- Add related hashtags at the end of the message."
+            "- Add related hashtags at the end of the message.",
+            "- Do not include the URL in the message.",
         ].joined(separator: "\n")
         Task {
-            let result = await Result { try await generativeContentRemote.generateText(siteID: siteID, base: prompt) }
+            let result = await Result { try await generativeContentRemote.generateText(siteID: siteID, base: prompt)
+                    .trimmingCharacters(in: CharacterSet(["\""]))  // Trims quotation mark
```
Comment (itsmeichigo) [inline Yosemite/Yosemite/Stores/ProductStore.swift:560]: Nit: I wonder if there's any edge case where quotation marks might be needed in the middle of sentences. Should we remove only the marks at the beginning and the end?

```suggestion
                    .removingPrefix("\"")
                    .removingSuffix("\"")
```

**Review by itsmeichigo — APPROVED** [review]

File: Yosemite/Yosemite/Stores/ProductStore.swift (hunk 1 above)
Comment (selanthiraiyan) [reply Yosemite/Yosemite/Stores/ProductStore.swift:560]: - `trimmingCharacters` only removes the characters in the `CharacterSet` from the beginning and end of the String. [Docs](https://developer.apple.com/documentation/foundation/nsstring/1415462-trimmingcharacters)
- We are also unit testing this behaviour [here](https://github.com/woocommerce/woocommerce-ios/pull/9921/files#diff-e646aaeb37f27f6e8e28316b94a04243eb7e834bf654550b4fcf2f825782dbe5R1876).

Kindly let me know if I am missing something here.

Comment (selanthiraiyan) [reply Yosemite/Yosemite/Stores/ProductStore.swift:560]: Merging the PR now. I will make changes in a separate PR if needed.

Comment (itsmeichigo) [reply Yosemite/Yosemite/Stores/ProductStore.swift:560]: My bad, please ignore my comment then.

Comment (selanthiraiyan) [reply Yosemite/Yosemite/Stores/ProductStore.swift:560]: No problem. Thanks again for the review! 🙇

**Pull request merged**